src/libsystemd-network/icmp6-util.c

```diff
@@ -83,6 +85,13 @@ int icmp6_bind_router_solicitation(int index) {
         if (r < 0)
                 return -errno;
 
+        if (!if_indextoname(index, ifname))
+                return -errno;
+
+        r = setsockopt(s, SOL_SOCKET, SO_BINDTODEVICE, ifname, strlen(ifname)+1);
```

Comment:
Hmm, I am puzzled by the "+1"... Does the socket option really require passing of the trailing NUL byte? It appears as if that's not necessary, and at least to me confusing.